### ISSUE TITLE
Can we ignore pytest for now? Or make it optional?

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/meta#50]
Currently we add a `test` environment to `tox.ini` which runs pytest.  But getting this to run properly is a lot of work.  And I don't want to just ignore it and leave it there, because then the next person (which may be me) will see it and expect `tox -e test` to work.  And I do want to "metaify" more Plone packages.  At our current rate, and without more people joining, it will take years.

Can we do something about that?
Some options from the top of my hat:

- Leave the tox env away always.
- Make it optional: command line option to either add it or leave it away, or an option in `.meta.toml`, like the tests/gha option, which is only mentioned in the docs now but no longer exists. :-)
- Use zope.testrunner for now, should be less work.

**Comment by gforcada** [comment]
Yes, sorry about that, I thought it would be much simpler to start using `pytest` and somehow, if anyway we want to move to it at some point, I thought that it might be easy to go through it 😅 

I was actually trying to use `zope.testrunner` on `plone.browserlayer` and I got it to work, but I had to move the source code inside a `src` folder, which is somewhat a best practice anyway 🤞🏾 🍀 

I was about to write an issue, but seems you were faster than me 😄 

With this, it should be straight forward to enable GHA to run the tests right away and have them always enabled on all repositories 👍🏾

**Comment by mauritsvanrees** [comment]
No need to be sorry. I hoped it would be easier as well.

**Comment by gforcada** [comment]
So, now that I slept a little bit 😅 

What I tested locally and it did work was (I will make a PR later):

`tox.ini`

```ini
[testenv:test]
usedevelop = true
deps =
    zope.testrunner
    -c https://dist.plone.org/release/6.0-dev/constraints.txt
commands =
    zope-testrunner --test-path=src -s plone.browserlayer
extras =
    test
```

And on `setup.py`

```py
    packages=find_packages('src'),
    package_dir={'': 'src'},
    namespace_packages=['plone'],
```

I'm always confused about these settings, but `z3c.form` for example is using them like this 🤷🏾 

`setuptools` [mentions this](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#src-layout)

**Comment by mauritsvanrees** [comment]
But then we would need to move to a src layout for all packages.  This probably means any existing PRs for packages will conflict because all files have been moved.  Maybe git is smart enough about this, but I don't count on that.

Can we auto discover this?  Maybe do an `os.listdir()` and see check for some directories like `plone`, `Products`, `collective`, `src`.
Maybe yet another option in `.meta.toml`...?

Also, we may need `--test-path={toxinidir}/src` to be sure.  I am never sure if within a `.tox/test` dir `tox` commands know in which directory they need to work.

**Comment by gforcada** [comment]
I wanted to try that, not moving to a src layout, yesterday, but I fall asleep 😅 both my kids are sick this week, so it's a bit exhausting 😕 

I guess it is possible, indeed, to not move to a `src` layout and get zope.testrunner to work, and all that without having to use `zc.buildout` to generate the scripts 🤞🏾 let's see if today I have time at night to look at it 🍀

**Comment by gforcada** [comment]
Bingo! 🍀 one does indeed not have to move the code to a `src` layout 🎉 

`zope-testrunner --test-path={toxinidir} -s plone.i18n`

Is enough to get it to work, I have it working on `plone.i18` (still not in the PR).

I will see how we can auto discover that on meta to adjust `tox.ini` as needed 😄

**Comment by mauritsvanrees** [comment]
Ah, adding `-s package.name` is indeed very useful, to prevent running tests for any other code that may be discovered in the top level.  As test I added a simple top level `tests` folder with `__init__.py` and a test file with a failing test, and without the `-s <package name>` this is indeed discovered, and the tests fail.
I created and merged PRs for plone.namedfile and plone.scale to fix this there as well.

- https://github.com/plone/plone.namedfile/pull/139
- https://github.com/plone/plone.scale/pull/73
